packages/compass-preferences-model/src/preferences.ts

```diff
@@ -139,6 +141,18 @@ function deriveNetworkTrafficOptionState<K extends keyof AllPreferences>(
       (v('networkTraffic') ? undefined : 'derived'),
   });
 }
+
+/** Helper for defining how to derive value/state for readOnly-affected preferences */
+function deriveReadOnlyOptionState<K extends keyof AllPreferences>(
+  property: K
+): DeriveValueFunction<boolean> {
+  return (v, s) => ({
+    value: v(property) && !v('readOnly'),
+    state:
+      s(property) ?? s('readOnly') ?? (v('readOnly') ? undefined : 'derived'),
```

Comment:
```suggestion
      s(property) ?? s('readOnly') ?? (!v('readOnly') ? undefined : 'derived'),
```

to match the networkTraffic option logic? Otherwise (as in the tests above) `enableShell: 'derived'` shows up when it shouldn’t and vice versa

Reply:
Yes, I am aware of this, fix is not pushed yet.